server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIndices.java

```diff
@@ -186,13 +233,27 @@ static final class Fields {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.field(Fields.COUNT, indexCount);
-        shards.toXContent(builder, params);
-        docs.toXContent(builder, params);
-        store.toXContent(builder, params);
-        fieldData.toXContent(builder, params);
-        queryCache.toXContent(builder, params);
-        completion.toXContent(builder, params);
-        segments.toXContent(builder, params);
+        if (shards != null) {
+            shards.toXContent(builder, params);
+        }
+        if (docs != null) {
```

Comment:
I think we should rely on test cases to validate the correctness of the response returned instead of coupling the null check verification with the requested metrics.

Reply:
Response needs to be constructed based on the user request and not based on transport response. It is better to throw errors when the expectation is not met rather than ignore silently.

Reply:
I agree its good to have those validation but the transport response is constructed based on user request. If a field is requested then the field will be set otherwise it won't be set, so I am not sure if we should validate this here again.

Reply:
If there is a bug in TransportAction and the response is not returned we would never catch the error.  We should assert the expectations of transport response at Rest layer and have it validated by IT .

Reply:
Will create a follow up git hub issue to address this.